app/validators/questions/calculated_question_validator.py

```diff
@@ -19,3 +22,38 @@ def validate_calculations(self):
             for answer_id in calculation["answers_to_calculate"]:
                 if answer_id not in answer_ids:
                     self.add_error(self.ANSWER_NOT_IN_QUESTION, answer_id=answer_id)
+
+    def _validate_answer_is_numeric(self, *, question_id, answer_id):
+        answer_type = self.schema.get_answer_type(answer_id)
+
+        if ANSWER_TYPE_TO_JSON_TYPE[answer_type.value] != TYPE_NUMBER:
+            self.add_error(
+                self.ANSWER_TYPE_FOR_CALCULATION_TYPE_INVALID.format(
+                    answer_type=ANSWER_TYPE_TO_JSON_TYPE[answer_type.value],
+                ),
+                referenced_question_id=question_id,
+                identifier=answer_id,
+            )
+
+    def validate_calculations_value_source_is_numeric(self):
+        """
+        Validates that source answer is of number type
+        """
+        for calculation in self.question.get("calculations"):
+            value = calculation.get("value")
+
+            if answer_id := calculation.get("answer_id"):
+                question_id = self.schema.get_block_id_by_answer_id(answer_id)
```

Comment:
Why is this getting the block?. What was wrong with the existing one?. We want the answer that is referencing it not the question the answer belongs to.

Reply:
As discussed, we scrap all these latest changes, the `question id` is part of the error message added automatically.